server/src/middlewares/HandleInvestigationRequest.ts

```diff
@@ -0,0 +1,62 @@
+import { NextFunction, Request, Response } from 'express';
+
+import logger from '../Logger/Logger';
+import UserType from '../Models/User/UserType';
+import { Severity } from '../Models/Logger/types';
+import { graphqlRequest } from '../GraphqlHTTPRequest';
+import { GET_INVESTIGATION_CREATOR } from '../DBService/InvestigationInfo/Query';
+
+export const handleInvestigationRequest = async (request: Request, response: Response, next: NextFunction) => {
+    const { user } = response.locals;
+    const epidemiologynumber = parseInt(response.locals.epidemiologynumber);
+
+    const InvestigationMiddlewareLogger = logger.setup({
+        workflow: 'InvestigationMiddleware',
+        investigation: +request.headers.epidemiologynumber,
+    });
+
+    if (isNaN(epidemiologynumber)) {
+        InvestigationMiddlewareLogger.info('client sent investigation request with no epidemiology number', Severity.MEDIUM);
+        return response.status(401).json({ error: 'no epidemiology number supplied' });
+    }
+
+    InvestigationMiddlewareLogger.info('getting current investigation details from DB', Severity.LOW);
+    const { investigationGroup, id , err} = await fetchInvestigationCreatorDetails(response.locals , epidemiologynumber , logger)
+    if(investigationGroup === -1) {
```

Comment:
Why didn’t you check the id as well?

Reply:
🤷‍♂️, I think that epidemiologyNumber will never be -1 so there's no need , however the id needs to be not null (typescript) , I'll look into it tomorrow maybe I can guarantee that the id is not null without giving it a value

Reply:
I've changed the error parameter [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1029/commits/565b8797549c53c61ff97ca956351a9afb4c20fc), it doesn't rely fake investigation numbers anymore 🚀